superset/assets/src/modules/colors.js

```diff
@@ -25,13 +23,9 @@ export const colorScalerFactory = function (colors, data, accessor, extents, out
     ext = extents;
   }
   if (data) {
-    ext = d3.extent(data, accessor);
+    ext = d3Extent(data, accessor);
   }
   const chunkSize = (ext[1] - ext[0]) / (colors.length - 1);
   const points = colors.map((col, i) => ext[0] + (i * chunkSize));
-  const scaler = d3.scale.linear().domain(points).range(colors).clamp(true);
-  if (outputRGBA) {
-    return v => hexToRGB(scaler(v));
-  }
-  return scaler;
+  return d3.scale.linear().domain(points).range(colors).clamp(true);
```

Comment:
just noting that if we swapped out `scaleLinear` from d3 v4 here would could get rid of d3 from another file -- def not blocking tho!

Reply:
There are slight difference between d3 v3 `scale.linear` and v4 `scaleLinear`. The `deck.gl` util that uses this function fails the test after the swap. I want to take a closer look at it in another PR and remove this `colorScalerFactory` function completely.